pkg/slack/filter.go

```diff
@@ -88,18 +86,56 @@ func NewFilter(filterLayers []string) (*MessageFilter, error) {
 	filter := MessageFilter{
 		whitelists: []messageFilterRule{},
 	}
-	// Ref: https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#create-a-check-run--parameters
+
+	used := struct{ keys []string }{keys: []string{}}
 	for _, layer := range filterLayers {
 		definition := strings.Split(layer, ":")
 
-		definition = definition
-		// switch definition[0]
-		// case ""
+		switch len(definition) {
+		case 1: // Assumed format "conclusion1,conclusion2,..."
+			rule := messageFilterRule{}
+			if slices.Contains(used.keys, "none") {
+				return nil, fmt.Errorf("duplicated conclusion strings; use commas to separate conclusions")
+			}
+			conclusions := strings.Split(definition[0], ",")
+
+			err := rule.setConclusions(conclusions)
+			if err != nil {
+				return nil, err
+			}
+
+			used.keys = append(used.keys, "none")
+			filter.whitelists = append(filter.whitelists, rule)
+		case 2, 3: // Assumed format "filterKey:filterValue1,filterValue2,..."
+			rule := messageFilterRule{}
+			filterType := definition[0]
+			if slices.Contains(used.keys, filterType) {
+				return nil, fmt.Errorf("duplicated filter type: %s", filterType)
+			}
+			switch filterType {
+			case "branches":
+				branches := strings.Split(definition[1], ",")
+				rule.Branches = branches
+			case "workflows":
+				workflows := strings.Split(definition[1], ",")
+				rule.Workflows = workflows
+			default:
+				return nil, fmt.Errorf("unsupported filter type: %s", filterType)
+			}
+
+			if len(definition) == 3 {
+				conclusions := strings.Split(definition[2], ",")
+
+				err := rule.setConclusions(conclusions)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			used.keys = append(used.keys, filterType)
+			filter.whitelists = append(filter.whitelists, rule)
```

Comment:
I suggest we introduce one more data model and we will do the parsing somewhere else so that when we call NewFilter, we don't need to deal with the parsing but just focusing on duplicated configs or other actual config validation.

Reply:
Bascially create MessageFilterRule somewhere and we will just accept message filter rules in NewFilter